compat/src/Children.js

```diff
@@ -1,9 +1,9 @@
 import { toChildArray } from 'preact';
 
-const mapFn = (children, fn) => {
+const mapFn = (children, fn, context) => {
 	if (!children) return null;
 	return toChildArray(children).reduce(
-		(acc, value, index) => acc.concat(fn(value, index)),
+		(acc, value, index) => acc.concat(fn(context, value, index)),
```

Comment:
That's a great start! Lookig more closely at the implementation in React it seems like the `context` arg is passed as a `thisArgument` and not passed as a "standard" argument in the classic sense:

```js
mapIntoArray(children, result, '', '', function(child) {
    return func.call(context, child, count++);
    //          ^^^^
});
```

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call) the signature of `Function.prototype.call()` is:

```js
func.call([thisArg[, arg1, arg2, ...argN]])
```

This means that the callback function will keep the same signature as before, so there should be no failing tests from this change.

```js
Children.map(children, (child, index) => ...);
```

 We'd only need a new one asserting that the `this` value inside the callback function is the same as  `context`.

Reply:
```suggestion
		(acc, value, index) => acc.concat(fn.call(context, value, index)),
```